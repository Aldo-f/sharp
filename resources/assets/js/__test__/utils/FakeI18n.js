export default {
    install() {
        window.i18n = {};
    }
}