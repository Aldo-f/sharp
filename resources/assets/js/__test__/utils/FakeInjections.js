import Vue from 'vue';

const injectedComponents = ['$field', '$form'];

export default {
    provide() {
        return injectedComponents.reduce((injections, prop) => {
            injections[prop] = new Vue();
            return injections;
        }, {});
    }
}