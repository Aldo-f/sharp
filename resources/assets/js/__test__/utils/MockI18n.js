import Localization ,* as localizationFn from '../../mixins/Localization';

const mockLangImplementation = localeKey => `{{ ${localeKey} }}`;

export default {
    install() {
        window.i18n = {};
    },

    mockLangFunction() {
        localizationFn.lang = jest.fn(mockLangImplementation);
        Localization.methods.l = jest.fn(mockLangImplementation);
    }
}