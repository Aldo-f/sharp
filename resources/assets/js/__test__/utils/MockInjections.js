import Vue from 'vue';

const injectedComponents = {
    $field:{},
    $form:{
        data: () => ({
            errors: {}
        })
    },
    $tab:{}
};

export default {
    provide() {
        return Object.keys(injectedComponents).reduce((injections, compName) => {
            let Comp = Vue.extend(injectedComponents[compName]);
            injections[compName] = new Comp();
            return injections;
        }, {});
    }
}