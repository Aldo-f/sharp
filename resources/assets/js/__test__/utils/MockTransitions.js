export default {
    install() {
        window.getComputedStyle = jest.fn(() => ({
            animationDelay:'',
            animationDuration:'',
            transitionDelay:'',
            transitionDuration:''
        }));
    }
}