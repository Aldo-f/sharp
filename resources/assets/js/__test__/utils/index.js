export { default as FakeInjections } from './FakeInjections';
export { default as FakeI18n } from './FakeI18n';
export { default as QueryComponent } from './QueryComponent';