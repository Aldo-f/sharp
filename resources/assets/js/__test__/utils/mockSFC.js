export function mockSFC (sfcModule, { template, render, ...extend }={}, mockOptions={}) {

    let { isJsFile } = mockOptions;

    if(isJsFile) {
        sfcModule.__oldDefault__ = { ...sfcModule.default }
    }

    Object.assign(sfcModule.default, {
        mixins: [],
        'extends': extend,
        template: template,
        render: !template ? render || function() { return this._v(`__MOCKED_SFC_${sfcModule.name}__\n`) } : null,
        watch: {},
        beforeCreate: ()=>{},
        created: ()=>{},
        beforeMount: ()=>{},
        mounted: ()=>{},
        beforeUpdate: ()=>{},
        updated: ()=>{},
        beforeDestroy: ()=>{},
        destroyed: ()=>{},
        data: ()=>({}),
        provide: {},
        inject: [],
    });

    //console.log(sfcModule);
    return sfcModule.default;
}

export function unmockSFC(sfcModule) {
    if(sfcModule.__oldDefault__) {
        sfcModule.default = sfcModule.__oldDefault__;
        return;
    }

    let { 'default':defaultObj, ...options } = sfcModule;
    sfcModule.default = options;
}