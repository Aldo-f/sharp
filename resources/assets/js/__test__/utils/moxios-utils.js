import moxios from 'moxios';

export function nextRequestFulfilled(config) {
    return new Promise(resolve => {
        moxios.wait(async () => {
            let request = moxios.requests.mostRecent();
            let response = await request.respondWith(config);
            resolve(response); // moxios Response object
        })
    });
}
