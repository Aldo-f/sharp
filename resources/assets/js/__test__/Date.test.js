import Vue from 'vue/dist/vue.common';
import DateField from '../components/form/fields/date/Date.vue';
import moment from 'moment';

import { FakeInjections, QueryComponent } from './utils';

describe('date-field',()=>{
    Vue.component('sharp-date', DateField);
    Vue.use(QueryComponent);

    beforeEach(() => {
        document.documentElement.lang = 'fr';
        document.body.innerHTML = `
            <div id="app">
                <sharp-date value="1996-08-20 12:11+02:00"
                            :has-date="!disableDate"
                            :has-time="!disableTime" 
                            :read-only="readOnly" 
                            :display-format="displayFormat"
                            :step-time="stepTime" 
                            :min-time="minTime" 
                            :max-time="maxTime"
                            @input="inputEmitted"
                            >
                </sharp-date>
            </div>
        `
    });

    it('can mount Date field', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can mount Date field without TimePicker', async () => {
        await createVm({
            propsData: {
                disableTime: true
            }
        });

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can mount "read only" Date field', async () => {
        await createVm({
            propsData: {
                readOnly: true,
                disableTime: true,
                disableDate: true
            }
        });

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can mount "picker displayed" Date field', async () => {
        let $date = await createVm({
            propsData: {
                disableTime: true,
                disableDate: true
            }
        });

        $date.showPicker = true;

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('expose appropriate props to TimePicker', async () => {
        let $date = await createVm({
            propsData: {
                displayFormat: 'HH : mm',
                stepTime: 10,
                minTime: '3:00',
                maxTime: '19:00'
            }
        });

        let timepicker = $date.$findChild('SharpTimePicker');

        $date.showPicker = true;

        await Vue.nextTick();

        expect(timepicker.$props).toMatchObject({
            value: {
                HH: '12',
                mm: '11',
                ss: '00'
            },
            active: true,
            format: 'HH : mm',
            minuteInterval: 10,
            min: '3:00',
            max: '19:00'
        })
    });

    it('expose appropriate props to DatePicker', async () => {
        let $date = await createVm();

        let datepicker = $date.$findChild('SharpDatepicker');

        expect(datepicker.$props).toMatchObject({
            value: new Date(1996, 7, 20, 12, 11),
            language: 'fr',
            inline: true,
            mondayFirst: true
        });
    });

    it('have correct input value', async () => {
        let $date = await createVm({
            propsData: {
                displayFormat: 'DD/MM/YYYY  HH : mm'
            }
        });

        expect($date.$refs.input.value).toBe('20/08/1996  12 : 11');
    });

    it('emit input on date changed & correct value', async () => {
        let $date = await createVm();

        let datepicker = $date.$findChild('SharpDatepicker');

        let inputEmitted = jest.fn();
        $date.$on('input', inputEmitted);
        datepicker.$emit('selected', new Date(1996, 7, 22));

        expect(inputEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted.mock.calls[0][0]).toBeInstanceOf(moment);
        expect(inputEmitted.mock.calls[0][0].toDate()).toEqual(new Date(1996, 7, 22, 12, 11));
    });

    it('emit input on time changed & correct value', async () => {
        let $date = await createVm();

        let timepicker = $date.$findChild('SharpTimePicker');

        let inputEmitted = jest.fn();
        $date.$on('input', inputEmitted);
        timepicker.$emit('change', { data: { HH: 13, mm: 20 } });

        expect(inputEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted.mock.calls[0][0]).toBeInstanceOf(moment);
        expect(inputEmitted.mock.calls[0][0].toDate()).toEqual(new Date(1996, 7, 20, 13, 20));
    });

    it('emit input on input changed', async () => {
        let $date = await createVm({
            displayFormat: 'DD/MM/YYYY HH:mm'
        });

        let { input } = $date.$refs;

        let inputEmitted = jest.fn();
        let okEmitted = jest.fn();

        $date.$on('input', inputEmitted);
        $date.$field.$on('ok', okEmitted);

        input.value = '22/08/1996 13:20';
        input.dispatchEvent(new Event('input', { bubbles: true }));

        expect(okEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted.mock.calls[0][0]).toBeInstanceOf(moment);
        expect(inputEmitted.mock.calls[0][0].toDate()).toEqual(new Date(1996, 7, 22, 13, 20));
    });

    it('emit error on input changed if invalid', async () => {
        let $date = await createVm({
            displayFormat: 'DD/MM/YYYY HH:mm'
        });

        let { input } = $date.$refs;

        let inputEmitted = jest.fn();
        let errorEmitted = jest.fn();

        $date.$field.$on('error', errorEmitted);
        $date.$on('input', inputEmitted);

        input.value = '20/08/1996 bug 12 : 40';
        input.dispatchEvent(new Event('input', { bubbles: true }));

        expect(errorEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted).toHaveBeenCalledTimes(0);
    });

    it('clear field state on blur', async () => {
        let $date = await createVm();

        let { input } = $date.$refs;

        let clearEmitted = jest.fn();

        $date.$field.$on('clear', clearEmitted);

        input.focus();
        input.blur();

        expect(clearEmitted).toHaveBeenCalled();
    });

    it('increase/decrease date on keydown up/down', async () => {
        let $date = await createVm({
            propsData: {
                displayFormat:'DD/MM/YYYY HH:mm'
            }
        });

        let { input } = $date.$refs;

        let inputEmitted = jest.fn();

        $date.$on('input', inputEmitted);

        input.setSelectionRange(1,1);
        input.dispatchEvent(new KeyboardEvent('keydown', { keyCode: 38 })); //UP

        expect(inputEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted.mock.calls[0][0]).toBeInstanceOf(moment);
        expect(inputEmitted.mock.calls[0][0].toDate()).toEqual(new Date(1997, 7, 20, 12, 10));
    });
});

async function createVm(customOptions={}) {
    let vm = new Vue({
        el: '#app',

        mixins: [customOptions, FakeInjections],
        props: ['readOnly', 'disableTime','disableDate' ,'displayFormat', 'stepTime', 'minTime', 'maxTime'],

        'extends': {
            methods: {
                inputEmitted:()=>{}
            }
        }
    });

    await Vue.nextTick();

    return vm.$children[0];
}