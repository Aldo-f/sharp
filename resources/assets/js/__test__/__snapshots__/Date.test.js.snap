// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-field can mount "picker displayed" Date field 1`] = `
<div id="app">
    <div class="SharpDate"><input class="SharpDate__input">
        <div class="SharpDate__picker" style="">
            <!---->
            <!---->
        </div>
    </div>
</div>
`;

exports[`date-field can mount "read only" Date field 1`] = `
<div id="app">
    <div class="SharpDate"><input disabled="disabled" class="SharpDate__input">
        <div class="SharpDate__picker" style="display: none;">
            <!---->
            <!---->
        </div>
    </div>
</div>
`;

exports[`date-field can mount Date field 1`] = `
<div id="app">
    <div class="SharpDate"><input class="SharpDate__input">
        <div class="SharpDate__picker" style="display: none;">
            <div class="SharpDate__datepicker SharpDate__date"><input type="hidden" readonly="readonly">
                <!---->
                <div class="SharpDate__calendar open" style="position: static;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-month">Août</span> <span class="SharpDate__cur-year">1996</span></span>
                        <span class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span></header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__weekdays"><span class="cell day-header">Lun</span><span class="cell day-header">Mar</span><span class="cell day-header">Mer</span><span class="cell day-header">Jeu</span><span class="cell day-header">Ven</span><span class="cell day-header">Sam</span>
                                <span
                                    class="cell day-header">Dim</span>
                            </div>
                            <div class="SharpDate__days">
                                <div class="SharpDate__dayContainer"><span class="cell day blank"></span><span class="cell day blank"></span><span class="cell day blank"></span> <span track-by="timestamp" class="cell day">1
                        </span><span track-by="timestamp" class="cell day">2
                        </span><span track-by="timestamp" class="cell day">3
                        </span><span track-by="timestamp" class="cell day">4
                        </span><span track-by="timestamp" class="cell day">5
                        </span><span track-by="timestamp" class="cell day">6
                        </span><span track-by="timestamp" class="cell day">7
                        </span><span track-by="timestamp" class="cell day">8
                        </span><span track-by="timestamp" class="cell day">9
                        </span><span track-by="timestamp" class="cell day">10
                        </span><span track-by="timestamp" class="cell day">11
                        </span><span track-by="timestamp" class="cell day">12
                        </span><span track-by="timestamp" class="cell day">13
                        </span><span track-by="timestamp" class="cell day">14
                        </span><span track-by="timestamp" class="cell day">15
                        </span><span track-by="timestamp" class="cell day">16
                        </span><span track-by="timestamp" class="cell day">17
                        </span><span track-by="timestamp" class="cell day">18
                        </span><span track-by="timestamp" class="cell day">19
                        </span><span track-by="timestamp" class="cell day selected">20
                        </span><span track-by="timestamp" class="cell day">21
                        </span><span track-by="timestamp" class="cell day">22
                        </span><span track-by="timestamp" class="cell day">23
                        </span><span track-by="timestamp" class="cell day">24
                        </span><span track-by="timestamp" class="cell day">25
                        </span><span track-by="timestamp" class="cell day">26
                        </span><span track-by="timestamp" class="cell day">27
                        </span><span track-by="timestamp" class="cell day">28
                        </span><span track-by="timestamp" class="cell day">29
                        </span><span track-by="timestamp" class="cell day">30
                        </span><span track-by="timestamp" class="cell day">31
                        </span> <span class="cell day blank"></span></div>
                            </div>
                        </div>
                    </div>
                </div>
                <div class="SharpDate__calendar" style="position: static; display: none;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-year">1996</span></span>
                        <span
                            class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span>
                    </header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__monthContainer"><span track-by="timestamp" class="cell month">Janvier</span><span track-by="timestamp" class="cell month">Février</span><span track-by="timestamp" class="cell month">Mars</span><span track-by="timestamp" class="cell month">Avril</span>
                                <span
                                    track-by="timestamp" class="cell month">Mai</span><span track-by="timestamp" class="cell month">Juin</span><span track-by="timestamp" class="cell month">Juillet</span><span track-by="timestamp" class="cell month selected">Août</span><span track-by="timestamp"
                                        class="cell month">Septembre</span><span track-by="timestamp" class="cell month">Octobre</span><span track-by="timestamp" class="cell month">Novembre</span><span track-by="timestamp" class="cell month">Décembre</span></div>
                        </div>
                    </div>
                </div>
                <div class="SharpDate__calendar" style="position: static; display: none;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-decade">1990's</span></span>
                        <span class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span></header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__yearContainer"><span track-by="timestamp" class="cell year">1990</span><span track-by="timestamp" class="cell year">1991</span><span track-by="timestamp" class="cell year">1992</span><span track-by="timestamp" class="cell year">1993</span>
                                <span
                                    track-by="timestamp" class="cell year">1994</span><span track-by="timestamp" class="cell year">1995</span><span track-by="timestamp" class="cell year selected">1996</span><span track-by="timestamp" class="cell year">1997</span><span track-by="timestamp" class="cell year">1998</span>
                                    <span
                                        track-by="timestamp" class="cell year">1999</span>
                            </div>
                        </div>
                    </div>
                </div>
            </div> <span class="time-picker SharpDate__time"><header><span class="hint">HH</span> <span class="hint">mm</span></header>
            <div class="dropdown">
                <div class="select-list">
                    <ul class="hours">
                        <li class="">00</li>
                        <li class="">01</li>
                        <li class="">02</li>
                        <li class="">03</li>
                        <li class="">04</li>
                        <li class="">05</li>
                        <li class="">06</li>
                        <li class="">07</li>
                        <li class="">08</li>
                        <li class="">09</li>
                        <li class="">10</li>
                        <li class="">11</li>
                        <li class="active">12</li>
                        <li class="">13</li>
                        <li class="">14</li>
                        <li class="">15</li>
                        <li class="">16</li>
                        <li class="">17</li>
                        <li class="">18</li>
                        <li class="">19</li>
                        <li class="">20</li>
                        <li class="">21</li>
                        <li class="">22</li>
                        <li class="">23</li>
                    </ul>
                    <ul class="minutes">
                        <li class="">00</li>
                        <li class="">30</li>
                    </ul>
                </div>
            </div>
            </span>
        </div>
    </div>
</div>
`;

exports[`date-field can mount Date field without TimePicker 1`] = `
<div id="app">
    <div class="SharpDate"><input class="SharpDate__input">
        <div class="SharpDate__picker" style="display: none;">
            <div class="SharpDate__datepicker SharpDate__date"><input type="hidden" readonly="readonly">
                <!---->
                <div class="SharpDate__calendar open" style="position: static;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-month">Août</span> <span class="SharpDate__cur-year">1996</span></span>
                        <span class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span></header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__weekdays"><span class="cell day-header">Lun</span><span class="cell day-header">Mar</span><span class="cell day-header">Mer</span><span class="cell day-header">Jeu</span><span class="cell day-header">Ven</span><span class="cell day-header">Sam</span>
                                <span
                                    class="cell day-header">Dim</span>
                            </div>
                            <div class="SharpDate__days">
                                <div class="SharpDate__dayContainer"><span class="cell day blank"></span><span class="cell day blank"></span><span class="cell day blank"></span> <span track-by="timestamp" class="cell day">1
                        </span><span track-by="timestamp" class="cell day">2
                        </span><span track-by="timestamp" class="cell day">3
                        </span><span track-by="timestamp" class="cell day">4
                        </span><span track-by="timestamp" class="cell day">5
                        </span><span track-by="timestamp" class="cell day">6
                        </span><span track-by="timestamp" class="cell day">7
                        </span><span track-by="timestamp" class="cell day">8
                        </span><span track-by="timestamp" class="cell day">9
                        </span><span track-by="timestamp" class="cell day">10
                        </span><span track-by="timestamp" class="cell day">11
                        </span><span track-by="timestamp" class="cell day">12
                        </span><span track-by="timestamp" class="cell day">13
                        </span><span track-by="timestamp" class="cell day">14
                        </span><span track-by="timestamp" class="cell day">15
                        </span><span track-by="timestamp" class="cell day">16
                        </span><span track-by="timestamp" class="cell day">17
                        </span><span track-by="timestamp" class="cell day">18
                        </span><span track-by="timestamp" class="cell day">19
                        </span><span track-by="timestamp" class="cell day selected">20
                        </span><span track-by="timestamp" class="cell day">21
                        </span><span track-by="timestamp" class="cell day">22
                        </span><span track-by="timestamp" class="cell day">23
                        </span><span track-by="timestamp" class="cell day">24
                        </span><span track-by="timestamp" class="cell day">25
                        </span><span track-by="timestamp" class="cell day">26
                        </span><span track-by="timestamp" class="cell day">27
                        </span><span track-by="timestamp" class="cell day">28
                        </span><span track-by="timestamp" class="cell day">29
                        </span><span track-by="timestamp" class="cell day">30
                        </span><span track-by="timestamp" class="cell day">31
                        </span> <span class="cell day blank"></span></div>
                            </div>
                        </div>
                    </div>
                </div>
                <div class="SharpDate__calendar" style="position: static; display: none;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-year">1996</span></span>
                        <span
                            class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span>
                    </header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__monthContainer"><span track-by="timestamp" class="cell month">Janvier</span><span track-by="timestamp" class="cell month">Février</span><span track-by="timestamp" class="cell month">Mars</span><span track-by="timestamp" class="cell month">Avril</span>
                                <span
                                    track-by="timestamp" class="cell month">Mai</span><span track-by="timestamp" class="cell month">Juin</span><span track-by="timestamp" class="cell month">Juillet</span><span track-by="timestamp" class="cell month selected">Août</span><span track-by="timestamp"
                                        class="cell month">Septembre</span><span track-by="timestamp" class="cell month">Octobre</span><span track-by="timestamp" class="cell month">Novembre</span><span track-by="timestamp" class="cell month">Décembre</span></div>
                        </div>
                    </div>
                </div>
                <div class="SharpDate__calendar" style="position: static; display: none;">
                    <header><span class="prev"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path></svg></span> <span class="up"><span class="SharpDate__cur-decade">1990's</span></span>
                        <span class="next"><svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd"><path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path></svg></span></header>
                    <div class="SharpDate__innerContainer">
                        <div class="SharpDate__rContainer">
                            <div class="SharpDate__yearContainer"><span track-by="timestamp" class="cell year">1990</span><span track-by="timestamp" class="cell year">1991</span><span track-by="timestamp" class="cell year">1992</span><span track-by="timestamp" class="cell year">1993</span>
                                <span
                                    track-by="timestamp" class="cell year">1994</span><span track-by="timestamp" class="cell year">1995</span><span track-by="timestamp" class="cell year selected">1996</span><span track-by="timestamp" class="cell year">1997</span><span track-by="timestamp" class="cell year">1998</span>
                                    <span
                                        track-by="timestamp" class="cell year">1999</span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <!---->
        </div>
    </div>
</div>
`;
